app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.plantsapp.presentation.ui.plantcreation
+
+import android.content.Context
+import android.net.Uri
+import com.example.plantsapp.presentation.ui.utils.createImageFile
+import com.example.plantsapp.presentation.ui.utils.getFileUri
+import dagger.hilt.android.qualifiers.ApplicationContext
+import timber.log.Timber
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import javax.inject.Inject
+
+class SaveUriInStorageUseCase @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+
+    fun saveImage(uri: Uri): SavingUriOutput {
+        val file = try {
+            val newFile = createImageFile(context) ?: throw IOException()
```

Comment:
I do not like the idea of throwing IOException by ourselves 

Questions:
- why createImageFile can return null? Maybe there some other way do it?
- get into documentation for File.createTempFile and maybe it always return non nullable? and issue is with java?

Reply:
okay, yeah, File.createTempFile() never returns null. it just throws an exception if something went wrong
so, yes, we can return non-null from createImageFile() and remove this null check